v2/dtos/provisionwatcher.go

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright (C) 2021 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package dtos
+
+import (
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+)
+
+// ProvisionWatcher and its properties are defined in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-metadata/2.x#/ProvisionWatcher
+type ProvisionWatcher struct {
+	common.Versionable  `json:",inline"`
+	Id                  string              `json:"id,omitempty" validate:"omitempty,uuid"`
+	Name                string              `json:"name" validate:"required,edgex-dto-none-empty-string"`
```

Comment:
Also need `edgex-dto-rfc3986-unreserved-chars` validation. Same for other names below.